tests/reftests/shared-fetch.test

```diff
@@ -0,0 +1,240 @@
+### sh mkurl.sh no-extra.3 sha256 sha512
+### sh mkurl.sh intra.3 sha512
+### sh mkurl.sh out.3 md5
+### opam update
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[default] synchronised from file://${BASEDIR}/REPO
+Now run 'opam upgrade' to apply any package updates.
+### opam remove no-extra
+The following actions will be performed:
+  - remove extra    3 [uses no-extra]
+  - remove no-extra 3
+===== 2 to remove =====
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   extra.3
+-> removed   no-extra.3
+Done.
+### opam install out
+The following actions will be performed:
+  - install no-extra 3 [required by extra]
+  - install extra    3 [required by intra]
+  - install intra    3 [required by out]
+  - install out      3
+===== 4 to install =====
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved extra.3, intra.3, no-extra.3, out.3  (cached)
+-> installed no-extra.3
+-> installed extra.3
+-> installed intra.3
+-> installed out.3
+Done.
+### opam remove no-extra
+The following actions will be performed:
+  - remove out      3 [uses intra]
+  - remove intra    3 [uses extra]
+  - remove extra    3 [uses no-extra]
+  - remove no-extra 3
+===== 4 to remove =====
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   out.3
+-> removed   intra.3
+-> removed   extra.3
+-> removed   no-extra.3
+Done.
+### <pkg:extra.3>
+opam-version: "2.0"
+build: [ "ocaml" "shared.ml" ]
+### <mkurl.sh>
+p=extra.3
+arch=shared
+file="REPO/packages/${p%.*}/$p/opam"
+echo "url {" >> $file
+echo "src: \"$arch.tgz\"" >> $file
+MD5="01234567890123456789012345678901"
+echo "checksum: \"md5=$MD5\"" >> $file
+echo "}" >> $file
+### sh mkurl.sh
+### opam update
+
+<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
+[default] synchronised from file://${BASEDIR}/REPO
+Now run 'opam upgrade' to apply any package updates.
+### opam install out | '[0-9a-f]{32,}' -> "a-given-hash"
+The following actions will be performed:
+  - install extra 3 [required by intra]
+  - install intra 3 [required by out]
+  - install out   3
+===== 3 to install =====
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+[ERROR] Conflicting file hashes, or broken or compromised cache!
+          - md5=a-given-hash (MISMATCH)
+          - md5=a-given-hash (match)
+          - sha512=a-given-hash (match)
+
+[ERROR] extra.3, intra.3, out.3: Checksum mismatch for file://${BASEDIR}/shared.tgz:
+          expected md5=a-given-hash
+          got      md5=a-given-hash
+[ERROR] Failed to get sources of extra.3, intra.3, out.3 (file://${BASEDIR}/shared.tgz): Checksum mismatch
+
+OpamSolution.Fetch_fail("Checksum mismatch")
+
+
+<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
++- The following actions failed
+| - fetch extra 3 3 3
```

Comment:
why is there `3 3 3` here? This seems a bit odd

Reply:
good catch!